spec/models/application_spec.rb

```diff
@@ -460,4 +460,32 @@ def subject.transient_previously_funded
       end
     end
   end
+
+  describe "#participant_outcome_state" do
+    subject { application.participant_outcome_state }
+
+    let(:application) { create(:application, :accepted, participant_outcome_state: "anything") }
+    let(:declaration) { create(:declaration, :completed, application:) }
+
+    before do
+      create(:participant_outcome, :failed, declaration:, created_at: 1.week.ago)
+      create(:participant_outcome, declaration:)
+    end
+
+    context "when ecf_api_disabled flag is toggled off" do
+      before { Flipper.disable(Feature::ECF_API_DISABLED) }
+
+      it "returns the attribute value" do
+        expect(subject).to eq("anything")
+      end
+    end
+
+    context "when ecf_api_disabled flag is toggled on" do
```

Comment:
Could we add more testing to cover possibly voided completed declarations which should be ignored, as well as have other declarations past completed? Also if there are no completed declarations as well shouldn't return anything?